### PR TITLE
Proxy API requests through the front end.

### PR DESCRIPTION
**Pull request by MancunianSam**
* Add controller endpoints to replace the API endpoints.
* Add a service to call the original API endpoints.
* Add tests for the new controller and service methods.
* Update the javascript to call the front end instead of the API
* Update the jest tests to work with the new fetch calls
* Remove apollo client and the graphql module.

### CODE REVIEW

File: app/controllers/UploadController.scala [hunk 1]
```diff
@@ -1,27 +1,56 @@
...
+  def startUpload(): Action[AnyContent] = secureAction.async { implicit request =>
+    request.body.asJson.flatMap(body => {
+      decode[StartUploadInput](body.toString).toOption
+    }) match {
+      case None => Future.failed(new Exception(s"Incorrect data provided ${request.body}"))
+      case Some(input) => uploadService.startUpload(input, request.token.bearerAccessToken).map(Ok(_))
+    }
+  }
+
+  def saveMetadata(): Action[AnyContent] = secureAction.async { implicit request =>
```
Comment (TomJKing) [inline app/controllers/UploadController.scala:45]: Maybe rename this: `saveClientMetadata`

File: app/controllers/UploadController.scala [hunk 2]
```diff
@@ -62,29 +91,29 @@ class UploadController @Inject()(val controllerComponents: SecurityComponents,
...
-    } yield {
-      val uploadStatus: Option[String] = consignmentStatus.flatMap(_.upload)
-      val pageHeading1stHalf = "Upload judgment"
-      val pageHeading2ndHalf = "Uploading judgment"
+      for {
+        consignmentStatus <- consignmentStatusService.getConsignmentStatus(consignmentId, request.token.bearerAccessToken)
+        reference <- consignmentService.getConsignmentRef(consignmentId, request.token.bearerAccessToken)
+      } yield {
+        val uploadStatus: Option[String] = consignmentStatus.flatMap(_.upload)
+        val pageHeading1stHalf = "Upload judgment"
```
Comment (TomJKing) [inline app/controllers/UploadController.scala:101]: Maybe rename these vals:

- `pageHeadingUpload`
- `pageHeadingUploading`

File: app/services/UploadService.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+  def startUpload(startUploadInput: StartUploadInput, token: BearerAccessToken): Future[String] = {
+    val variables = su.Variables(startUploadInput)
+    sendApiRequest(startUploadClient, su.document, token, variables).map(data =>
+      data.startUpload
+    )
+  }
+
+  def saveMetadata(addFileAndMetadataInput: AddFileAndMetadataInput, token: BearerAccessToken): Future[List[afam.AddFilesAndMetadata]] = {
```
Comment (TomJKing) [inline app/services/UploadService.scala:36]: Maybe rename: `saveClientMetadata`

File: test/services/UploadServiceSpec.scala [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+
+class UploadServiceSpec extends AnyFlatSpec {
+  implicit val ec: ExecutionContext = ExecutionContext.global
+  private val graphQlConfig = mock[GraphQLConfiguration]
+  private val token = new BearerAccessToken("some-token")
+
+  "startUpload" should "returns the correct data" in {
```
Comment (TomJKing) [inline test/services/UploadServiceSpec.scala:25]: Reword slightly: `... return the correct data`

File: test/services/UploadServiceSpec.scala [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    val graphQlResponse =
+      GraphQlResponse(Some(su.Data("ok")),Nil)
+    when(graphQlClientForStartUpload.getResult(token, su.document, Some(su.Variables(input))))
+      .thenReturn(Future.successful(graphQlResponse))
+
+    val response = new UploadService(graphQlConfig).startUpload(input, token).futureValue
+    response should equal("ok")
+  }
+
+  "saveMetadata" should "returns the correct data" in {
```
Comment (TomJKing) [inline test/services/UploadServiceSpec.scala:40]: Reword slightly: `... return the correct data`

File: app/services/UploadService.scala [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    val variables = ucs.Variables(consignmentStatusInput)
+    sendApiRequest(updateConsignmentStatusClient, ucs.document, token, variables).map(data => {
+      data.updateConsignmentStatus match {
+        case Some(response) => response
+        case None => throw new RuntimeException(s"No data returned when updating the consignment status for ${consignmentStatusInput.consignmentId}")
+      }
+    })
+  }
+
+  def startUpload(startUploadInput: StartUploadInput, token: BearerAccessToken): Future[String] = {
```
Comment (TomJKing) [inline app/services/UploadService.scala:29]: Is it possible to have no response here? If so, should an error be thrown?

**Review by TomJKing — COMMENTED** [review]
Looks good. Some minor comments.

Also when testing locally seeing the following error when trying to upload a judgment. Standard uploads working OK.

Screenshot from 2022-07-11 07-49-43

**Comment by MancunianSam** [comment]
I only tested this for a standard user and forgot there's different views for the judgment users. The CSRF input was missing so I've added that in and added in a missing entry to the routes file which was causing an error further on. You can now do a complete judgment export.

File: app/services/UploadService.scala (hunk 6 above)
Comment (MancunianSam) [reply app/services/UploadService.scala:29]: The `ApiErrorHandling` object calls `result.data.get` so if data was optional, we'd get a (slightly ugly) error and it also throws a `GraphQlException` if the API populates the errors field. I think that covers everything that could happen. What do you think?

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**